lib/active_merchant/billing/gateways/firstdata_e4_v27.rb

```diff
@@ -0,0 +1,445 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class FirstdataE4V27Gateway < Gateway
+      self.test_url = 'https://api.demo.globalgatewaye4.firstdata.com/transaction/v27'
+      self.live_url = 'https://api.globalgatewaye4.firstdata.com/transaction/v27'
+
+      TRANSACTIONS = {
+        sale:          '00',
+        authorization: '01',
+        verify:        '05',
+        capture:       '32',
+        void:          '33',
+        credit:        '34',
+        store:         '05'
+      }
+
+      SUCCESS = 'true'
+
+      SENSITIVE_FIELDS = [:cvdcode, :expiry_date, :card_number]
+
+      BRANDS = {
```

Comment:
I'm confused about the split between `BRANDS` and `E4_BRANDS`. [We only use `BRANDS` for `supported_cardtypes`](https://github.com/activemerchant/active_merchant/pull/2912/files#diff-8cf7977aecc61a9b94a9aa5bd7c9a55aR29), which initially made me think that `E4_BRANDS` was totally redundant, but then [we do use `E4_BRANDS` for card types](https://github.com/activemerchant/active_merchant/pull/2912/files#diff-8cf7977aecc61a9b94a9aa5bd7c9a55aR320) later on. I _think_ we're doing this because it was a backwards-compatibility kluge inherited from the original Firstdata gateway; if that's the case, then can we kill it, since we're introducing a specific v27 upgrade here? I think we could just have `BRANDS` and be done.

Reply:
Resolved this by killing `E4_BRANDS`.